locations/method_locations.go

```diff
@@ -48,3 +48,8 @@ func MethodOperationInfo(m *desc.MethodDescriptor) *dpb.SourceCodeInfo_Location
 func MethodSignature(m *desc.MethodDescriptor, index int) *dpb.SourceCodeInfo_Location {
 	return pathLocation(m, 4, int(apb.E_MethodSignature.TypeDescriptor().Number()), index) // MethodDescriptor.options == 4
 }
+
+// MethodOption returns the precise location of the method's option with the given field number, if any.
+func MethodOption(m *desc.MethodDescriptor, fieldNumber int) *dpb.SourceCodeInfo_Location {
```

Comment:
Could you also change `MethodHTTPRule` and `MethodOperationInfo` above to use `MethodOption`?